RELEASING.md

```diff
@@ -19,10 +83,19 @@ This script takes optional arguments denoting the new versions for each qualifie
   git add .
   git commit -m "Release [VERSION HERE]"
   ```
-  
+
+1. Make sure you are using Maven version 3.3 or higher to support the Nexus plugin required to stage a release.
+
+1. To ensure a clean build, remove *all* Maven targets (including subdirectories not handled by `mvn clean`) by running `rm -rf target */target`.
+
 5. Run `utilities/stage_release.sh`.
 This script builds and stages the release artifact on the Maven Central Repository, updates the README.md files with the release version + commits them locally, and finally generates a new site version for the gh-pages branch under a temporary directory named `tmp_gh-pages`. If you haven't run the release process before, it's worth verifying everything; check the staged release on the Sonatype website, and verify that the local commits have the right version updates.
 
+If you experience failures, you may need to:
+- repeat the clean step above
+- remove the temporary directory created to store docs by running `rm -rf tmp_gh-pages`
+- remove staged repositories from Sonatype: if a staged repository appears [here](https://oss.sonatype.org/#nexus-search;quick~com.google.cloud), remove it by running `mvn nexus-staging:drop`.
```

Comment:
I think the staged repo would not cause the staging to fail, it will just have both artifacts staged at the same time. I think it is worth noting that users should always check it they have something staged already before doing this step.